plugin/trino-duckdb/src/main/java/io/trino/plugin/duckdb/DuckDbClient.java

```diff
@@ -0,0 +1,306 @@
+import static io.trino.plugin.jdbc.StandardColumnMappings.smallintWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
+import static io.trino.plugin.jdbc.TypeHandlingJdbcSessionProperties.getUnsupportedTypeHandling;
+import static io.trino.plugin.jdbc.UnsupportedTypeHandling.CONVERT_TO_VARCHAR;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoField.EPOCH_DAY;
+
+public final class DuckDbClient
+        extends BaseJdbcClient
+{
+    private static final Pattern DECIMAL_PATTERN = Pattern.compile("DECIMAL\\((?<precision>[0-9]+),(?<scale>[0-9]+)\\)");
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd");
+
+    @Inject
+    public DuckDbClient(
+            BaseJdbcConfig config,
+            ConnectionFactory connectionFactory,
+            QueryBuilder queryBuilder,
+            IdentifierMapping identifierMapping,
+            RemoteQueryModifier queryModifier)
+    {
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, queryModifier, false);
+    }
+
+    @Override
+    public Connection getConnection(ConnectorSession session)
+            throws SQLException
+    {
+        // The method calls Connection.setReadOnly method, but DuckDB does not support changing read-only status on connection level
+        return connectionFactory.openConnection(session);
+    }
+
+    @Override
+    public void renameSchema(ConnectorSession session, String schemaName, String newSchemaName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+    }
+
+    @Override
+    protected String escapeObjectNameForMetadataQuery(String name, String escape)
+    {
+        // org.duckdb.DuckDBDatabaseMetaData.getSearchStringEscape returns null
+        checkArgument(escape == null, "escape must be null");
+        return name;
+    }
+
+    @Override
+    protected Optional<List<String>> getTableTypes()
+    {
+        return Optional.of(ImmutableList.of("BASE TABLE", "LOCAL TEMPORARY", "VIEW"));
```

Comment:
Do we need to expose local temporary tables via Trino ?

Reply:
#24847